samples/cellular/nrf_cloud_multi_service/README.rst

```diff
@@ -819,25 +817,25 @@ See the :file:`src/prj.conf` configuration file and the :file:`overlay_nrf7002ek
 
 This overlay also enables the :ref:`TLS Credentials Shell <zephyr:tls_credentials_shell>` for run-time credential installation.
 
-If you are certain you understand the risks, you can configure your build to use Wi-Fi connectivity on the nRF5340 DK with the nRF7002 EK shield by using the ``--board nrf5340dk_nrf5340_cpuapp_ns`` target and the ``-DSHIELD=nrf7002ek`` and ``-DOVERLAY_CONFIG=overlay_nrf7002ek_wifi_no_lte.conf`` options.
-
-You must also configure a (globally unique) device ID at build time by enabling the :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_COMPILE_TIME` Kconfig option and setting :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID` to the device ID.
-
-For example, for a device with the device ID ``698d4c11-0ccc-4f04-89cd-6882724e3f6f``:
+If you are certain you understand the risks, you can configure your build to use Wi-Fi connectivity on the nRF5340 DK with the nRF7002 EK shield by using the following variables:
 
-.. tabs::
-
-   .. group-tab:: Bash
-
-      .. code-block:: console
-
-        west build --board nrf5340dk_nrf5340_cpuapp_ns -p always -- -DSHIELD=nrf7002ek -DOVERLAY_CONFIG=overlay_nrf7002ek_wifi_no_lte.conf -DCONFIG_NRF_CLOUD_CLIENT_ID_SRC_COMPILE_TIME=y -DCONFIG_NRF_CLOUD_CLIENT_ID="698d4c11-0ccc-4f04-89cd-6882724e3f6f"
-
-   .. group-tab:: PowerShell
-
-      .. code-block:: console
-
-         west build --board nrf5340dk_nrf5340_cpuapp_ns -p always -- -DSHIELD=nrf7002ek "-DOVERLAY_CONFIG=overlay_nrf7002ek_wifi_no_lte.conf" -DCONFIG_NRF_CLOUD_CLIENT_ID_SRC_COMPILE_TIME=y  "-DCONFIG_NRF_CLOUD_CLIENT_ID=\`"698d4c11-0ccc-4f04-89cd-6882724e3f6f\`""
+.. list-table::
+  :header-rows: 1
+
+  * - Build target
+    - Files and Kconfig options to use
+    - Build variables
+  * - ``nrf5340dk_nrf5340_cpuapp_ns``
+    - | * :file:`overlay_nrf7002ek_wifi_no_lte.conf`
+      | * :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_COMPILE_TIME`
+      | * :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID` (set to the globally unique device ID)
+    - | * ``--board nrf5340dk_nrf5340_cpuapp_ns``
+      | * ``-DEXTRA_CONF_FILE=overlay_nrf7002ek_wifi_no_lte.conf``
+      | * ``-DSHIELD=nrf7002ek``
+      | * ``-DCONFIG_NRF_CLOUD_CLIENT_ID_SRC_COMPILE_TIME=y``
+      | * ``-DCONFIG_NRF_CLOUD_CLIENT_ID="698d4c11-0ccc-4f04-89cd-6882724e3f6f"`` (for ID ``698d4c11-0ccc-4f04-89cd-6882724e3f6f``)
```

Comment:
I echo the same sentiment as @gregersrygg. It is very helpful to have concrete examples of some of the valid combinations of compiler arguments written directly in the samples. In fact, I myself frequently reference these. It would be detrimental, in my opinion, to force anyone reading these to construct the full commands themselves based on the list of options they need to enable.

Reply:
100% agreed on behalf of modem_shell and location samples. I know some of my team mates are using our documentation to copy paste the commands for various variants of our samples/apps. Using build_target was already breaking this a bit. For any new developers, this PR would make things a lot harder.

The tables are definitely a nice addition. I haven't thought about this enough to have an opinion what we could do to ease the maintenance of the docs but I think we cannot remove all example commands from samples/apps.

Reply:
> please check the PR again after edits have been applied.

I'll add my comment about the table and command example here.
The table still looks somewhat distracting to me. Not having enough time to look at this deeply, I've tried to figure out how to make it better in my opinion. One problem currently is that you still don't have a very easy way of copy pasting the command or its parameters. You need to copy several parameters and if you take them from the 3rd column, they require formatting (removing line breaks) or copying each parameter separately.

I've been thinking about the columns and the fact that 2nd and 3rd columns have almost the same information. Here is what I would consider doing:

1. Make 3rd column the string of all arguments after the `--` in `west build` command. Not bulleted list, no line breaks. It will be wrapped into several lines by the browser but copy pasting still works as a one liner.
For example: `-DEXTRA_CONF_FILE="overlay-cloud_mqtt.conf;overlay-pgps.conf" -DCONFIG_LOCATION_SERVICE_EXTERNAL=y -DCONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE=y`
2. Do we need 2nd column? The same information is in 3rd column. I'm not sure about this item as I guess it's for VS Code users, which I've never used for compiling our code. This is more to provoke discussion.

The ultimate test for this document change would be to give the document to someone who hasn't compiled before asking him/her to provide a compilation command for modem shell with P-GPS configuration 😄 And we could give him/her a link to the modem shell table created in this PR.

Reply:
Agreed, largely, with @trantanen, I think the tables are useful enough to someone who is experienced with our toolchain, but if I were an absolute beginner, I would find myself very confused.

As things stand right now, I'd be forced to click through several links and make several inferences -- And the tendency to need to do that in our documentation is already something I struggled with as a beginner when I first on-boarded into NCS. So I really hesitate to require even more of that.

I do see the value in the table in that it provides a semantic understanding of the parameters that are being used. But I also think a copy-and-pastable west build command is invaluable to beginners.

Would one solution perhaps be to generate the table along with an example command-line snippet on a single line underneath the table?

Reply:
> Would one solution perhaps be to generate the table along with an example command-line snippet on a single line underneath the table?

Funny that I actually tried to search a bit if we had an option to do selections from a list and then generate text based on that. We could have selection of a feature and another selection of a build target. However, I couldn't find such elements from RST but maybe documentation team can check this again. I guess this might be doable with some extra work such as adding Javascript or some other language somehow to the document.